1_fetch.R

```diff
@@ -86,6 +87,25 @@ p1_targets_list <- list(
   tar_target(
     p1_reaches_sf,
     st_read(p1_reaches_shp)
+  ),
+  
+  # Download NHDPlusV2 flowlines for DRB
+  tar_target(
+    p1_nhdv2reaches_sf,
+    {
+      drb_huc8s %>%
+        # For each HUC8 watershed, retrieve NHDPlusV2 flowlines
+        purrr::map(.,get_nhdv2_flowlines) %>%
```

Comment:
Curious why you are using `purrr` instead of `targets` dynamic branching.

Reply:
If you go through the branching approach, might also be easier to put this process into a separate function and then apply dynamic branching. I am not sure if it would work better, but just a suggestion.

Reply:
Great points - it does seem like there might be a way to simplify these lines. I had to remind myself that I ended up using `purrr` here for two reasons:  
1) you can only branch over named targets and not global objects. So to branch over `drb_huc8s`, that object would need to be a target. 
2) `nhdplusTools::get_nhdplus` requires an AOI, but seems to have a cap for the number of flowlines that get returned (maybe this is 10,000?). So the population of NHD reaches was getting truncated when I tried to `get_nhdplus()` over the entire DRB. To get around that, I split the DRB into the contributing HUC8 watersheds, and applied the `get_nhdplus` function over each HUC8.